app/connectors/FinancialInstitutionsConnector.scala

```diff
@@ -47,12 +47,16 @@ class FinancialInstitutionsConnector @Inject() (val config: FrontendAppConfig, v
       .get(url"${config.fIManagementUrl}/crs-fatca-fi-management/financial-institutions/$subscriptionId/$fiId")
       .execute[HttpResponse]
 
-  def addFi(fiDetails: CreateFIDetails)(implicit
+  def addOrUpdateFI(fiDetails: CreateFIDetails, requestType: String)(implicit
     hc: HeaderCarrier,
     ec: ExecutionContext
-  ): Future[Either[ErrorDetails, HttpResponse]] =
-    httpClient
-      .post(url"${config.fIManagementUrl}/crs-fatca-fi-management/financial-institutions/create")
+  ): Future[Either[ErrorDetails, HttpResponse]] = {
+    val request = if (requestType == "POST") {
```

Comment:
Instead of a HardCoded `POST` string, better to use a `sealed trait` like `CheckMode` we usually use?